packages/app/src/debug/DebugFailedTest.cy.tsx

```diff
@@ -1,37 +1,195 @@
 import DebugFailedTest from './DebugFailedTest.vue'
+import type { TestResults } from './DebugSpec.vue'
+
+const group1 = {
+  os: {
+    id: '123',
+    name: 'Linux',
+    nameWithVersion: 'Linux Debian',
+  },
+  browser: {
+    id: '123',
+    formattedName: 'Chrome',
+    formattedNameWithVersion: 'Chrome 106',
+  },
+  groupName: 'Staging',
+  id: '123',
+}
+
+const group2 = {
+  os: {
+    id: '123',
+    name: 'Windows',
+    nameWithVersion: 'Windows 110',
+  },
+  browser: {
+    id: '123',
+    formattedName: 'Electron',
+    formattedNameWithVersion: 'Electron 106',
+  },
+  groupName: 'Production',
+  id: '456',
+}
+
+const rowElementTesting = (testResult: TestResults) => {
+  const l = testResult.titleParts.length
+  const finalPartLength = testResult.titleParts[l - 1].length
+  const assertionArr = [{ testAttr: 'failed-icon', text: '' }]
+
+  if (l <= 3) {
+    testResult.titleParts.forEach((title, index) => {
+      if (index === l - 1) {
+        assertionArr.push({ testAttr: `titleParts-${index}-title`, text: testResult.titleParts[l - 1].slice(0, finalPartLength - 15) })
+        assertionArr.push({ testAttr: `titleParts-${index + 1}-title`, text: testResult.titleParts[l - 1].slice(finalPartLength - 15) })
+      } else {
+        assertionArr.push({ testAttr: `titleParts-${index}-title`, text: title })
+        assertionArr.push({ testAttr: `titleParts-${index + 1}-chevron`, text: '' })
+      }
+    })
+  } else {
+    testResult.titleParts.forEach((title, index) => {
+      if (index === l - 1) {
+        if (testResult.titleParts[length - 1]) {
+          assertionArr.push({ testAttr: `titleParts-${index + 1}-chevron`, text: '' })
+          assertionArr.push({ testAttr: `titleParts-${index + 1}-title`, text: testResult.titleParts[l - 1].slice(0, finalPartLength - 15) })
+          assertionArr.push({ testAttr: `titleParts-${index + 2}-title`, text: testResult.titleParts[l - 1].slice(finalPartLength - 15) })
+        }
+      } else {
+        if (index === 0) {
+          assertionArr.push({ testAttr: `titleParts-${index}-title`, text: title })
+          assertionArr.push({ testAttr: `titleParts-${index + 1}-chevron`, text: '' })
+          assertionArr.push({ testAttr: `titleParts-1-title`, text: '...' })
+        } else {
+          assertionArr.push({ testAttr: `titleParts-${index + 1}-chevron`, text: '' })
+          assertionArr.push({ testAttr: `titleParts-${index + 1}-title`, text: title })
+        }
+      }
+    })
+  }
+
+  cy.findByTestId('test-row').children().each((ele, index) => {
+    if (assertionArr[index]) {
+      cy.wrap(ele).should('have.attr', 'data-cy', assertionArr[index].testAttr)
+      if (assertionArr[index].text !== '') {
+        cy.wrap(ele).should('contain.text', assertionArr[index].text)
+      }
+    }
+  })
```

Comment:
I think this can be made easier to follow, something like:

```suggestion
  cy.get('[data-cy*=titleParts]').each((ele, index) => {
    if (index >= assertionArr.length) {
      // we don't assert about all the title parts every time
      return
    }

    const { testAttr, text } = assertionArr[index]

    cy.findByTestId(testAttr).should('contain.text', text)
  })
```

This would avoid the `children` and the need to wrap elements. Only drawback is a harmless assertion that the svgs have empty string text.